src/types.rs

```diff
@@ -170,7 +170,29 @@ impl fmt::Display for ValidatorStatus {
     }
 }
 
-#[derive(Debug)]
+impl FromStr for ValidatorStatus {
+    type Err = &'static str;
```

Comment:
nit: formatting
```suggestion
    type Err = &'static str;
    
```

Reply:
also it may end up being more convenient for callers if the error type is just a `String`